Cargo.toml

```diff
@@ -0,0 +1,5 @@
+# We make the whole Synapse folder a workspace so that we can run `cargo`
+# commands from the root (rather than having to cd into rust/).
+
+[workspace]
+members = ["rust"]
```

Comment:
I wonder whether we should split things out into multiple crates later on, notably because the unit of compilation in Rust is a crate.
Definitely something we can afford to do at a later point in time, so no blocker, just inviting conversation.

Reply:
Yeah, possibly. Though for dev builds incremental compilation is now a thing.